run-lcb-evals.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+# This script runs the LiveCodeBench custom evaluator for each of the models
+# listed in `MODELS`.
+#
+# It assumes that post-processed data in the format expected by the LiveCodeBench
+# custom evaluator: https://github.com/LiveCodeBench/LiveCodeBench?tab=readme-ov-file#custom-evaluation)
+# is stored in the following directory structure:
+#
+# data/
+# ├─ gpt-4.1-2025-04-14/
+# │  ├─ post-processed/
+# │  │  ├─ lcb-formatted.json
+# ├─ Llama-3.3-70B-Instruct/
+# │  ├─ post-processed/
+# │  │  ├─ lcb-formatted.json
+#
+# This script produces a set of LiveCodeBench evaluation files for each
+# LiveCodeBench input file in the same directory where it is located.
+#
+# Ensure you are running this script in a directory that also contains the `data`
+# folder at the same depth.
+#
+# Example usage:
+#   ./run-lcb-evals.py
+
+from dataclasses import dataclass
+
+import glob
+import subprocess
+import concurrent.futures
+
+
+@dataclass
+class LcbModelEvaluationInfo:
+    """Represents an LLM and its LiveCodeBench input files.
+
+    Attributes:
+        model (str): An LLM.
+        lcb_input_files (list[str]): A list of `.json` files conforming to the
+            format expected by the LiveCodeBench custom evaluator.
+            See: https://github.com/LiveCodeBench/LiveCodeBench?tab=readme-ov-file#custom-evaluation
+    """
+
+    model: str
+    lcb_input_files: list[str]
+
+
+def main() -> None:
+    model_to_lcb_input_files: dict[str, list[str]] = {
+        model: glob.glob(f"data/{model}/post-processed/*lcb-formatted.json") for model in MODELS
+    }
+    model_evaluation_infos: list[LcbModelEvaluationInfo] = [
+        LcbModelEvaluationInfo(model=model, lcb_input_files=lcb_inputs)
+        for model, lcb_inputs in model_to_lcb_input_files.items()
+    ]
+
+    with concurrent.futures.ThreadPoolExecutor(max_workers=3) as executor:
```

Comment:
I suspect you can afford more workers than 3, because each process has to read and write files, which is relatively slow.  (OK, I retract that.  It's true for a hard drive, but not true for a SSD.  But still, you should be able to afford as many cores or threads as the CPU has.)